### PR TITLE
Add support for duplicating lines

### PR DESCRIPTION
**Pull request by scholtzan**
Duplicates lines that are selected

### CODE REVIEW

File: rust/core-lib/src/editor.rs [hunk 1]
```diff
@@ -687,6 +688,38 @@ impl Editor {
         }
     }
 
+    fn duplicate_line(&mut self, view: &View, config: &BufferItems) {
+        let mut builder = delta::Builder::new(self.text.len());
+        // get affected lines
+        let mut lines = BTreeSet::new();
```
Comment (scholtzan) [inline rust/core-lib/src/editor.rs]: This part is almost the same as in `modify_indent`. Perhaps there is a way to reuse the code...

Comment (cmyr) [reply rust/core-lib/src/editor.rs]: that could make sense at some point, feels like this is okay for now though.

**Comment by scholtzan** [comment]
I added it under "Other movement and deletion commands". Not sure if that is the right section though.

**Comment by nangtrongvuon** [comment]
I noticed Travis has been building this branch for a while. Maybe something got stuck?

**Comment by cmyr** [comment]
hmm, good point. I checked and it said it passed, just this graphic hasn't updated?

**Comment by eyelash** [comment]
After duplicating multiple lines the selection is somewhat unexpected:

before | after
:-:|:-:
![before][before] | ![after][after]

This makes it hard to run `duplicate_line` multiple times.

[before]: https://user-images.githubusercontent.com/2668288/45583828-44fa6d00-b8c9-11e8-8c53-d10822463256.png
[after]: https://user-images.githubusercontent.com/2668288/45583831-4e83d500-b8c9-11e8-871f-44a9b76a858b.png

**Comment by cmyr** [comment]
hmm good point, I think I'd call that a bug. @scholtzan?

**Comment by scholtzan** [comment]
Oh, I will look into it.

**Pull request merged**